evap/results/tests/test_exporters.py

```diff
@@ -254,3 +255,41 @@ def test_contributor_result_export(self):
         self.assertEqual(workbook.sheets()[0].row_values(8)[2], 3.0)
         self.assertEqual(workbook.sheets()[0].row_values(10)[0], "Overall Average Grade")
         self.assertEqual(workbook.sheets()[0].row_values(10)[2], 3.25)
+
+    def test_text_answer_export(self):
+        evaluation = baker.make(Evaluation, name_en="eval_en", name_de="eval_de")
+        questions = [
+            baker.make(Question, questionnaire__type=10),
+            baker.make(Question, questionnaire__type=20),
+            baker.make(Question, questionnaire__type=30),
+        ]
+        text_answers = [
+            baker.make(TextAnswer, question=questions[0], contribution__evaluation=evaluation),
+            baker.make(TextAnswer, question=questions[1], contribution__evaluation=evaluation),
+            baker.make(TextAnswer, question=questions[2], contribution__evaluation=evaluation),
+            baker.make(TextAnswer, question=questions[2], contribution__evaluation=evaluation),
+            baker.make(TextAnswer, question=questions[0], contribution__evaluation=evaluation),
+        ]
+
+        results = defaultdict(list)
+        for text_answer in text_answers:
+            question_text = text_answer.question
+            results[question_text].append(text_answer.answer)
+
+        binary_content = BytesIO()
+        TextAnswerExcelExporter(evaluation.name, evaluation.course.semester.name,
+                                evaluation.course.responsibles_names,
+                                results).export(binary_content)
+        binary_content.seek(0)
+        workbook = xlrd.open_workbook(file_contents=binary_content.read())
+        sheet = workbook.sheets()[0]
+
+        # Sheet headline
+        self.assertEqual(sheet.row_values(0)[0], evaluation.name)
+        self.assertEqual(sheet.row_values(1)[0], evaluation.course.semester.name)
+        self.assertEqual(sheet.row_values(2)[0], evaluation.course.responsibles_names)
+
+        # Questions, ordered by questionnaire type (answers must not be ordered)
```

Comment:
"(answers must not be ordered)" is supposed to mean "The excel-importer should use the order given by the results list", right? If so, I'd change this to:
```python
# Questions, ordered by questionnaire type. Answers keep their order from results list
```
or similar. What do you think?

Reply:
Since the questions are fetched once when rendering the site and again when generating the Excel File, I am not sure if the answers actually are in the same order. Further, the order of the answers is not tested here (or anywhere else).
Still, this comment is more accurate, since the exporter does not change the order, so I will adapt it.